### DISCUSSION TITLE
routeRule cache with custom key

### DISCUSSION THREAD

**Discussion by niddu85** [nitrojs/nitro#1170] [Q&A]
Hello,
i'm trying to figure out how to cache a route (eg the homepage of my website) but based on custom key.

this is because the language of the website is calculated based on the URL (eg: www.mywebsite.com is english, fr.mywebsite.com is for french, de.mywebsite.com is for german...) but the nuxt page that manages it is always the index.vue file, giving different HTML results depending on the language

what i want to do is something like:

```
routeRules: {
    "/": {
      swr: 60 * 60,
      cache: {
        base: "redis",
        getKey: () => return myCustomKey   <-----
      },
    },
  },
```

I don't know if am I following the right way or if maybe there is a different approach to this

Thanks in advance

**Comment by William-WFC** [comment]
I have the same confusion. I guess this is a bug. getKey seems not conversion right.
https://github.com/unjs/nitro/issues/1660

**Comment by lxccc812** [comment]
Is there any answer?

**Comment by pi0** [comment]
Hi please followup via https://github.com/unjs/nitro/issues/1048
